cmd/recorder/utils.go

```diff
@@ -158,3 +161,97 @@ func (ct *clientTransport) RoundTrip(req *http.Request) (*http.Response, error)
 
 	return res, err
 }
+
+func getInsecureOrigins(siteURL string) ([]string, error) {
+	if siteURL == "" {
+		return nil, fmt.Errorf("invalid siteURL: should not be empty")
+	}
+
+	var insecureOrigins []string
+
+	u, err := url.Parse(siteURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse SiteURL: %w", err)
+	}
+	// If the given SiteURL is not running on a secure HTTPs connection, we add
+	// it to the list of allowed insecure origins. We assume this would only
+	// happen either on internal private networks or for development/testing purposes.
+	if u.Scheme == "http" {
+		insecureOrigins = []string{
+			siteURL,
```

Comment:
Can we trust that the siteURL is always an origin, i.e. that there's no path? And if we can't, how does Chromium behave when you give it a full URL instead of an origin?

Reply:
Well, we are trusting the MM side to pass something here which should be an origin. Of course if that side is compromised they could pass anything. But the origin portion is what counts so `<scheme>://<hostname>:<port>`. If a path is passed on top of that then it's essentially ignored. As long as the origin part matches, then opening a URL with that origin will be treated as secure. Chromium will start even if we pass garbage to it but of course it won't be treating as secure anything insecure at that point.